central/policy/service/service_impl.go

```diff
@@ -1122,3 +1125,52 @@ func combineStrings(toCombine [][]string) []string {
 	}
 	return nil
 }
+
+func (s *serviceImpl) SaveAsCustomResources(ctx context.Context, request *v1.SaveAsCustomResourcesRequest) (*v1.SaveAsCustomResourcesResponse, error) {
+	policyList, missingIndices, err := s.policies.GetPolicies(ctx, request.GetPolicyIds())
+	if err != nil {
+		return nil, err
+	}
+	errDetails := &v1.PolicyOperationErrorList{}
+	for _, missingIndex := range missingIndices {
+		policyID := request.GetPolicyIds()[missingIndex]
+		errDetails.Errors = append(errDetails.Errors, &v1.PolicyOperationError{
+			PolicyId: policyID,
+			Error: &v1.PolicyError{
+				Error: "not found",
+			},
+		})
+		log.Warnf("A policy error occurred for id %s: not found", policyID)
+	}
+	if len(missingIndices) > 0 {
+		statusMsg, err := status.New(codes.InvalidArgument, "Some policies could not be retrieved. Check the error details for a list of policies that could not be found").WithDetails(errDetails)
+		if err != nil {
+			return nil, utils.ShouldErr(errors.Errorf("unexpected error creating status proto: %v", err))
+		}
+		return nil, statusMsg.Err()
+	}
+
+	errDetails = &v1.PolicyOperationErrorList{}
+	resp := new(v1.SaveAsCustomResourcesResponse)
+	for _, policy := range policyList {
+		cr, err := customresource.GenerateCustomResource(policy)
+		if err != nil {
+			errDetails.Errors = append(errDetails.Errors, &v1.PolicyOperationError{
+				PolicyId: policy.GetId(),
+				Error: &v1.PolicyError{
+					Error: errors.Wrap(err, "Failed to marshal policy to custom resource").Error(),
+				},
+			})
+		}
+		resp.CustomResources = append(resp.CustomResources, cr)
+	}
+	if len(errDetails.GetErrors()) > 0 {
+		statusMsg, err := status.New(codes.InvalidArgument, "Some policies could not be marshal to custom resources. Check the error details for the list of policies").WithDetails(errDetails)
```

Comment:
Failed to save all policies as custom resources. Check error details for a list of policies.